[test]: spanner it test failure

Spanner it test on main failing since [this pr](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/pull/3995) merged to main

https://github.com/GoogleCloudPlatform/spring-cloud-gcp/actions/runs/16684046522/job/47229243062

```
Error: 0:609 [ERROR]   SpannerRepositoryIntegrationTests.queryMethodsTest_deleteAllById_doesNothingOnEmptyIds:227 » Spanner INVALID_ARGUMENT: io.grpc.StatusRuntimeException: INVALID_ARGUMENT: Failed to initialize transaction due to invalid mutation key.
```

--- a/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/repository/support/SimpleSpannerRepository.java
+++ b/spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/repository/support/SimpleSpannerRepository.java
@@ -176,6 +176,9 @@ public class SimpleSpannerRepository<T, I> implements SpannerRepository<T, I> {
   @Override
   public void deleteAllById(Iterable<? extends I> ids) {
     Assert.notNull(ids, "IDs must not be null");
+    if (!ids.iterator().hasNext()) {
+      return;
+    }
     KeySet.Builder builder = KeySet.newBuilder();
     for (Object id : ids) {
       builder.addKey(toKey(id));

